subsys/bluetooth/host/conn.c

```diff
@@ -2408,11 +2403,6 @@ int bt_conn_le_phy_update(struct bt_conn *conn,
 {
 	uint8_t phy_opts, all_phys;
 
-	if (IS_ENABLED(CONFIG_BT_AUTO_PHY_UPDATE) &&
-	    !atomic_test_bit(conn->flags, BT_CONN_AUTO_PHY_COMPLETE)) {
-		return -EAGAIN;
-	}
-
```

Comment:
I like the idea of this PR, but in the case that the application attempts to do a `bt_conn_le_phy_update` before the auto-initiated PHY update has completed (i.e. before `le_phy_update_complete` has been called), doesn't removing such a check like this cause issues?

Reply:
Yes, this should be kept. If the AUTO_PHY (or the others) are not completed the application should not be able to start these. I think there might be an issue with updating the internal state of the conn object. In addition this would likely result in an HCI error code as well.

Reply:
I had a thought about that. Before this PR:
```
	if (IS_ENABLED(CONFIG_BT_AUTO_PHY_UPDATE) &&
	    !atomic_test_bit(conn->flags, BT_CONN_AUTO_PHY_COMPLETE)) {
		return -EAGAIN;
	}
```
would result in an error code. Now 
```
	return bt_le_set_phy(conn, all_phys, param->pref_tx_phy,
			     param->pref_rx_phy, phy_opts);
```
would return an error.  

As both code paths will return an error, I thought that it was better to simplify the code. Feel free to disagree.

Reply:
@rugeGerritsen  Can you elaborate on when/why `bt_le_set_phy` will return an error if the stack is doing a PHY update when the application calls `bt_conn_le_phy_update`? I can't find it

Reply:
`bt_conn_le_phy_update()` issues the command to the Link Layer. Some link layers may simply accept the command, others may return "BUSY". For both cases, I believe the application gets what it wants. Either two phy update callbacks, or only one and an error when calling `bt_conn_le_phy_update()`

Reply:
Ah, so this is an intentional change in host behavior? 

The HCI spec does not disallow this, so that's fine with me, but please add this part in your commit message, so it is clear that is a known change in behavior (i.e. allowing more than 1 outstanding PHY update at a time) when both the stack and the application attempts to change the PHY at the same time.

Reply:
Yes, this was intentional. Let me update the commit message if @joerchan and others agree

Reply:
I won't object, but maybe check with @alwa-nordic @seko-nordic as well?

Reply:
@alwa-nordic seems to approve it as well :)